File: umbrella-insurance-react-ui/src/models/records/teamRecords/v1/TeamRecord.ts
import { Season } from "../../../seasons/v1/Season";
import { Team } from "../../../teams/v1/Team";
import { Record } from "../../v1/Record";

export class TeamRecord {
    id?: number;
    team?: Team;
    record?: Record;
    season?: Season;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.team) {
                this.team = new Team(obj.team);
            }
            if(obj?.record) {
                this.record = new Record(obj.record);
            }
            if(obj?.season) {
                this.season = new Season(obj.season);
            }
        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/TeamRecord.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "team_records", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "team_records_unique", columnNames = {"season_id", "team_id"})
})
public class TeamRecord {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "team_record_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "record_id")
    private Record record;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "team_id")
    private Team team;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "season_id")
    private Season season;

}